lib/validation/validator.js

```diff
@@ -91,6 +93,27 @@ async function _validate(schemaName, options) {
 	await schemaValidator.validate(options);
 }
 
+async function _validateAndSetDefaults(schemaName, options) {
+	if (!defaultsValidator[schemaName]) {
+		defaultsValidator[schemaName] = (async () => {
+			const {default: Ajv} = await import("ajv");
+			const {default: ajvErrors} = await import("ajv-errors");
+			return new Validator({Ajv, ajvErrors, ajvConfig: {useDefaults: true}, schemaName});
+		})();
+	}
+
+	// When AJV is configured with useDefaults: true, it may add properties to the
+	// provided configuration that were not initially present. This behavior can
+	// lead to unexpected side effects and potential issues. To avoid these
+	// problems, we create a copy of the configuration. If we need the altered
+	// configuration later, we return this copied version.
+	const optionsCopy = {...options};
```

Comment:
I suppose we should create a deep copy? Or are we only concerned about root level changes?

Reply:
Thanks!